### ISSUE TITLE
OmniLight3D nodes render improperly after reaching greater than 44 shadow-enabled sources in a scene

### ISSUE THREAD

**Issue by the-sink** [godotengine/godot#64557]
### Godot version

4.0.alpha14

### System information

Windows 10, Vulkan Clustered, RTX 2060 (516.94)

### Issue description

If there are greater than 44 (?) OmniLights with shadows enabled in a 3D scene, some will begin to attenuate improperly. It also appears to be partly influenced by the range of the light source, and the camera position, with more lights acting strangely as their range is decreased.

Lights will render properly until turning shadows on. Once turned on, the visual bug will appear on some lights and will not disappear even if shadows are then turned off, until the editor has been restarted. Here's a video demonstrating what I'm seeing:
https://www.youtube.com/watch?v=vL94p9ZuduA

The project shown in the video is also attached below.

I'm unsure if this is the result of an intentional engine limit, but the fact that seemingly random lights have attenuation issues while not fully disappearing, and it not returning to normal after disabling shadows, leads me to assume it's a bug.

### Steps to reproduce

- Create a new project
- Add some geometry to be able to see the lights (in the attached project a simple plane mesh was used)
- Begin adding OmniLight3D nodes to the scene (at least 45)
- Enable shadows on all lights
- Observe some lights failing to attenuate properly

### Minimal reproduction project

[lighting-bug.zip](https://github.com/godotengine/godot/files/9367370/lighting-bug.zip)

**Comment by Calinou** [comment]
Can you reproduce this after adjusting the shadow quadrant values in the project settings? Instead of the default 4/4/16/64 split, try something like 16/16/16/64.

**Comment by the-sink** [comment]
Yeah, that definitely helps in regards to the number of light sources that render properly. At 16/16/16/256 for example, I can get up to 152 shadow-enabled lights before new ones start to break (ignoring the performance implications of course...)
